### PR TITLE
GEOPY-832

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-832 - Update pre-commits**

### CODE REVIEW

**Review by benk-mira — CHANGES_REQUESTED** [review]
Couple of comments and one potential bug to check out

File: geoh5py/objects/object_base.py [hunk 1]
```diff
@@ -350,7 +340,7 @@ def get_data(self, name: str | uuid.UUID) -> list[Data]:
 
         return entity_list
 
-    def get_data_list(self, attribute="name") -> list[str]:
+    def get_data_list(self, attribute="name") -> list:
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoh5py/objects/object_base.py:343]: Done

File: geoh5py/objects/object_base.py [hunk 2]
```diff
@@ -398,23 +388,25 @@ def property_groups(self) -> list[PropertyGroup] | None:
...
+        if isinstance(property_groups, PropertyGroup):
+            property_groups = [property_groups]
 
-            for prop_group in prop_groups:
-                if not any(
-                    pg.uid == prop_group.uid for pg in property_groups
-                ) and not any(pg.name == prop_group.name for pg in property_groups):
-                    prop_group.parent = self
-                    property_groups += [prop_group]
+        if self.property_groups is None:
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoh5py/objects/object_base.py:397]: Done

File: geoh5py/objects/object_base.py [hunk 3]
```diff
@@ -398,23 +388,25 @@ def property_groups(self) -> list[PropertyGroup] | None:
...
+            return
+
+        keepers = []
+        for property_group in self.property_groups:
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoh5py/objects/object_base.py:401]: Other way around, but yes agree

File: geoh5py/objects/object_base.py [hunk 4]
```diff
@@ -398,23 +388,25 @@ def property_groups(self) -> list[PropertyGroup] | None:
...
 
-    @property_groups.setter
-    def property_groups(self, prop_groups: list[PropertyGroup]):
-        if prop_groups is None:
-            property_groups = None
-        else:
-            property_groups = self._property_groups
-            if property_groups is None:
-                property_groups = []
+    def remove_property_groups(
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoh5py/objects/object_base.py:391]: It removes the property group as listed by the input?

[the remark this answers is not in the document]
Comment (benk-mira) [reply geoh5py/objects/object_base.py:391]: yes, you're right.

**Review by benk-mira — APPROVED** [review]
all good

**Pull request merged**